libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -743,6 +746,10 @@ protected virtual void OnSetScopedServices(DialogContext dialogContext)
             if (Generator != null)
             {
                 dialogContext.Services.Set(this.Generator);
+                if (this.Generator is ResourceMultiLanguageGenerator)
+                {
+                    Expression.Functions.Add(MissingPropertiesFunction.Name, new MissingPropertiesFunction(dialogContext));
+                }
```

Comment:
1) Is this the right place to do this?  Isn't the logical place to have the generator register it?
2) This seems unduly specific to ResourceMultiLanguageGenerator.
A better way of doing this would be to add a method to LanguageGenerator to allow any derivative of that class to register services/functions.

Reply:
Previous comment still applies.  How would someone create another generator that added a function?